packages/sanity/src/core/field/diff/components/ChangeBreadcrumb.tsx

```diff
@@ -1,17 +1,24 @@
-import React from 'react'
+import React, {useMemo} from 'react'
 import {Breadcrumbs, Text} from '@sanity/ui'
 import {ChangeTitlePath, FieldChangeNode} from '../../types'
 import {ChangeTitleSegment} from './ChangeTitleSegment'
 
 /** @internal */
 export function ChangeBreadcrumb(props: {change?: FieldChangeNode; titlePath: ChangeTitlePath}) {
   const {change, titlePath} = props
+  const maxLength = useMemo(() => {
+    if (change && change?.titlePath.length > 3) {
+      return 4
+    }
+    return undefined
+  }, [change])
 
   return (
     <Breadcrumbs
+      maxLength={maxLength}
```

Comment:
💭 instead of a function here, can't we always define a `maxLength=4` ? Since it will only trigger that if the items are greater than the maxLength. 

Eg: https://www.sanity.io/ui/arcade?mode=jsx&jsx=eJyzccqvUMhNLErPzLOtNqm141KAAhunotTElOSi0tykYrigAlBphU9qXnpJBkg1knhxakFiUWJJfpFtNZIo0JiQ1IoShdzSktQUheLMqlTbakMkSyCgWslOqRZVlz5IG7I6hDyyqE1KZpmdR2pOTr6CoY0%2BiINd0gifpDE%2BSRN8kqb4JM3wSZqjSdroIwU3TBgomF9hxwXjjciYwhMwAHr1qQ0%3D